client/components/domains/register-domain-step/index.jsx

```diff
@@ -320,6 +320,23 @@ class RegisterDomainStep extends React.Component {
 
 	render() {
 		const queryObject = getQueryObject( this.props );
+		let i, ReactComponentName;
+		if ( this.state.notice ) {
+			// Fix restoring components from Redux state.
+			for ( i = 0; i < this.state.notice.length; i++ ) {
+				if (
+					typeof this.state.notice[ i ].type === 'string' &&
+					typeof this.state.notice[ i ].$$typeof === 'undefined'
```

Comment:
This works, but it'd be nice to understand why it's happening in the first place. I think if we could fix the root cause, we wouldn't have to rely on checking `$$type` or other React internal properties. Maybe @jsnajdr has a few insights?

I did a spot of testing. For the hard refresh related to this issue, the [madness kicks off in](https://github.com/Automattic/wp-calypso/blob/master/client/components/domains/register-domain-step/index.jsx#L293) `componentDidUpdate()`.

My general assumption is that, for whatever reason, the React component passed to `translate()` loses its way in the render stack. So as you've shown, it will work in the context of `render()`. I think that's why this solution also does the job: https://github.com/Automattic/wp-calypso/compare/master...fix/signup-domains-notice

In my version, I did a comparison on the `message` [object](https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/registration/availability-messages.js/#L267) that `getAvailabilityNotice()` returns. For more reasons I cannot fathom, I suspect React Fiber's reconciliation is working to ensure our notice message is recognized as a fragment and the tree renders correctly. This, however, is also conjecture.

**Before:**
<img width="896" alt="screen shot 2018-05-06 at 1 30 44 pm" src="https://user-images.githubusercontent.com/6458278/39670003-3b7852cc-513d-11e8-945e-d9aaccacb425.png">

**After (see __owner_):**
<img width="920" alt="screen shot 2018-05-06 at 2 31 59 pm" src="https://user-images.githubusercontent.com/6458278/39670002-387ad84c-513d-11e8-8497-25cfa9b6354e.png">

This component could do with a bit of attention general, for example, cutting down the redraws, and perhaps reviewing what's triggered in the lifecycle methods.

Reply:
If I understand the issue correctly, we store a value returned from `translate` in `this.state.notice`. Its type is `PropTypes.node`, i.e., something that can be returned from a `render` function or passed as `children`. In this case, it's an array of strings or JSX elements.

The issue description says that this state is stored in Redux and loaded from local storage on refresh. That would certainly break things, because the serialization loses parts of the object, namely the `Symbol` value of `$$typeof`.

But I don't see `this.state.notice` stored or retrieved from Redux anywhere?

Anyway, the best solution seems to be to not store JSX elements in state or Redux, but store the primitive data values (domain, site, error) instead and create the JSX markup in `render`. That's exactly what @ramonjd `fix/signup-domains-notice` does.

Reply:
Thanks for your insight. Happy to close this PR unmerged, I was just trying to attempt recovering things for users stuck in this. Maybe they are not many, though.

Reply:
I've created an alt PR #24720